code/modules/surgery/organs/organ_internal.dm

```diff
@@ -343,16 +343,41 @@
 	parent_organ = "head"
 	slot = "eyes"
 	var/list/eye_colour = list(0,0,0)
+	var/list/colourmatrix = null
+	var/list/colourblind_matrix = null //Special colourblindness parameters.
+	var/colourblind_darkview = null
+	var/dependent_disabilities = null //Gets set by eye-dependent disabilities such as colourblindness so the eyes can transfer the disability during transplantation.
+	var/dark_view = 2 //Default dark_view for Humans.
+	var/weld_proof = null //If set, the eyes will not take damage during welding. eg. IPC optical sensors do not take damage when they weld things while all other eyes will.
 
 /obj/item/organ/internal/eyes/proc/update_colour()
 	dna.write_eyes_attributes(src)
 
-/obj/item/organ/internal/eyes/insert(mob/living/carbon/M, special = 0)
+/obj/item/organ/internal/eyes/proc/get_colourmatrix() //Returns a colour matrix if the eyes are organic.
+	return !robotic && owner.disabilities & COLOURBLIND ? (colourblind_matrix ? colourblind_matrix : MATRIX_GREYSCALE) : colourmatrix //Returns special colourmatrix if colourblind and it exists and greyscale if it doesn't, otherwise returns current velue.
```

Comment:
The if-then blocks aren't reeeally that large, especially if I set MATRIX_GREYSCALE as the default value for `colourblind_matrix`, but for maintainability's sake